lib/reply.js

```diff
@@ -627,7 +643,7 @@ function onSendEnd (reply, payload) {
     return
   }
 
-  if ((statusCode >= 100 && statusCode < 200) || statusCode === 204) {
+  if ((statusCode >= 100 && statusCode < 200) || statusCode === 204 || statusCode === 304) {
```

Comment:
Does this count as a bugfix/feat or a breaking change?

Reply:
Bug fix, it was broken before.

Actually I started with this fix I was returning 304 and it was not working, but 204 was.

Then noticed the other problems: any `Response.body === null` would never end the reply and headers were not properly set in the body-less responses (including 204)